src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -525,6 +580,381 @@ public List<UserType> fetchUserTypes() {
+    }
+
+    public CaseWorkerProfile processExistingCaseWorkers(
+            StaffProfileCreationRequest cwUiRequest, CaseWorkerProfile caseWorkerProfiles) {
+
+        CaseWorkerProfile filteredProfile = null;
+
+        if (isTrue(caseWorkerProfiles.getSuspended())) {
+            //when existing profile with delete flag is true then log exception add entry in exception table
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE, ALREADY_SUSPENDED_ERROR_MESSAGE,
+                    StringUtils.EMPTY, cwUiRequest, STAFF_PROFILE_UPDATE);
+            throw new InvalidRequestException(ALREADY_SUSPENDED_ERROR_MESSAGE);
+        } else if (cwUiRequest.isSuspended()) {
+            //when existing profile with delete flag is true in request then suspend user
+            if (isUserSuspended(UserProfileUpdatedData.builder().idamStatus(IDAM_STATUS_SUSPENDED).build(),
+                    caseWorkerProfiles.getCaseWorkerId(), ORIGIN_EXUI, cwUiRequest.getRowId())) {
+                caseWorkerProfiles.setSuspended(true);
+                filteredProfile = caseWorkerProfiles;
+                return filteredProfile;
+            }
+        } else {
+            //when existing profile with delete flag is false then update user in CRD db and roles in SIDAM
+            filteredProfile = updateSidamRoles(caseWorkerProfiles, cwUiRequest);
+            return filteredProfile;
+        }
+        //add user roles in user profile and filter out UP failed records
+
+        return null;
+    }
+
+
+    public CaseWorkerProfile persistCaseWorker(
+            CaseWorkerProfile updateCaseWorkerProfile,
+            StaffProfileCreationRequest cwUiRequest) {
+        List<CaseWorkerProfile> processedCwProfiles = null;
+        List<CaseWorkerProfile> profilesToBePersisted = new ArrayList<>();
+
+        profilesToBePersisted.addAll(deleteChildrenAndUpdateCwProfiles(updateCaseWorkerProfile, cwUiRequest));
+        //profilesToBePersisted.add(suspendedCaseWorkerProfiles);
+        profilesToBePersisted = profilesToBePersisted.stream().filter(Objects::nonNull).collect(toList());
+
+        if (isNotEmpty(profilesToBePersisted)) {
+            processedCwProfiles = caseWorkerProfileRepo.saveAll(profilesToBePersisted);
+            log.info("{}:: {} case worker profiles inserted ", loggingComponentName,
+                    processedCwProfiles.size());
+        }
+
+        CaseWorkerProfile caseWorkerProfile = null;
+
+        if (!ObjectUtils.isEmpty(processedCwProfiles)) {
+            caseWorkerProfile = processedCwProfiles.get(0);
+        }
+        return caseWorkerProfile;
+    }
+
+    // deletes children and updates caseworker profile
+    private List<CaseWorkerProfile> deleteChildrenAndUpdateCwProfiles(CaseWorkerProfile updateCaseWorkerProfiles,
+                                                                      StaffProfileCreationRequest cwUiRequest) {
+        List<CaseWorkerProfile> updatedProfiles = new ArrayList<>();
```

Comment:
You need not need a list as you will be updating only one caseworker

Reply:
done